components/compositions/compositions.service.ts

```diff
@@ -0,0 +1,267 @@
+    private HsMapService: HsMapService,
+    private HsCore: HsCoreService,
+    private HsCompositionsParserService: HsCompositionsParserService,
+    private HsConfig: HsConfig,
+    private HsPermalinkUrlService: HsShareUrlService,
+    private HsUtilsService: HsUtilsService,
+    private HsStatusManagerService: HsStatusManagerService,
+    private HsCompositionsMickaService: HsCompositionsMickaService,
+    private HsCompositionsStatusManagerMickaJointService: HsCompositionsStatusManagerMickaJointService,
+    private HsCompositionsLaymanService: HsCompositionsLaymanService,
+    private $log: HsLogService,
+    private $window: Window,
+    private HsCommonEndpointsService: HsCommonEndpointsService,
+    private HsCompositionsMapService: HsCompositionsMapService,
+    private HsEventBusService: HsEventBusService
+  ) {
+    this.tryParseCompositionFromCookie();
+    this.tryParseCompositionFromUrlParam();
+    if (HsPermalinkUrlService.getParamValue('permalink')) {
+      this.parsePermalinkLayers();
+    }
+
+    this.HsEventBusService.mapResets.subscribe(() => {
+      this.HsCompositionsParserService.composition_loaded = null;
+      this.HsCompositionsParserService.composition_edited = false;
+    });
+
+    this.HsEventBusService.compositionEdits.subscribe(() => {
+      this.HsCompositionsParserService.composition_edited = true;
+    });
+
+    this.HsEventBusService.compositionLoadStarts.subscribe((id) => {
+      id = `${this.HsStatusManagerService.endpointUrl()}?request=load&id=${id}`;
+      this.HsCompositionsParserService.loadUrl(id);
+    });
+
+    this.HsEventBusService.vectorQueryFeatureSelection.subscribe((e) => {
+      const record = this.HsCompositionsMapService.getFeatureRecordAndUnhighlight(
+        e.feature,
+        e.selector
+      );
+      if (record) {
+        this.loadComposition(this.getRecordLink(record));
+      }
+    });
+  }
+
+  datasetSelect(id_selected) {
+    this.data.id_selected = id_selected;
+  }
+
+  async loadCompositions(ds, params) {
+    this.HsCompositionsMapService.clearExtentLayer();
+    const bbox = this.HsMapService.getMapExtentInEpsg4326();
+    await this.managerByType(ds).loadList(
+      ds,
+      params,
+      bbox,
+      this.HsCompositionsMapService.extentLayer
+    );
+  }
+
+  resetCompositionCounter() {
+    this.HsCommonEndpointsService.endpoints.forEach((ds) => {
+      if (ds.type == 'micka') {
+        this.HsCompositionsMickaService.resetCompositionCounter(ds);
+      }
+    });
+  }
+
+  managerByType(endpoint) {
+    switch (endpoint.type) {
+      case 'micka':
+        return this.HsCompositionsStatusManagerMickaJointService;
+      case 'layman':
+        return this.HsCompositionsLaymanService;
+      default:
+        this.$log.warn(`Endpoint type '${endpoint.type} not supported`);
+    }
+  }
+
+  deleteComposition(composition) {
+    const endpoint = composition.endpoint;
+    this.managerByType(endpoint).delete(endpoint, composition);
```

Comment:
Yes, it will fail. I was counting that delete dialog would never be called for other endpoint types, and jsut leaving the default switch to get rid of linting errors, but yeah, you can never be sure what will be added in future.